src/cpp/fastdds/xtypes/serializers/json/dynamic_data_json.cpp

```diff
@@ -744,9 +746,20 @@ ReturnCode_t json_serialize_basic_member(
             if (RETCODE_OK == ret)
             {
                 // Insert UTF-8 converted value
+#if defined(MINGW_COMPILER)
+                std::wstring aux_wstring_value({value});
+                std::string utf8_value;
+                int size_needed = std::wcstombs(nullptr, aux_wstring_value.data(), 0);
+                if (size_needed > 0) {
```

Comment:
You forget this change. Also, don't add empty spaces after the line
```suggestion
                if (size_needed > 0)
                {
```